src/main/java/org/resitrack/controller/PersonController.java

```diff
@@ -51,13 +94,76 @@ public void searchPeopleByName() {
         }
     }
 
+    /**
+     * This function is used to delete one or many people
+     *
+     * @param name
+     * @return true if delete process success and vice versa
+     */
+    public boolean deletePeopleByName(String name) {
+        List<Person> peopleByName = findPersonByName(name);
+        if (peopleByName.size() < 0) {
+            return false;
+        }
+        return deletePeople(peopleByName);
+    }
+
+    /**
+     * This function is used to show basic information of one person
+     *
+     * @param person
+     */
     public void printPersonInformation(Person person) {
         System.out.println("Id: " + person.getId());
         System.out.println("Name: " + person.getName());
         System.out.println("Gender: " + person.getGender());
         System.out.println("Dob: " + person.getDob());
     }
 
+    /**
+     * This function is used to delete many people
+     *
+     * @param people
+     * @return true if delete people success and vice versa
+     */
+
+    private boolean deletePeople(List<Person> people) {
+        List<Person> deletePeople = new ArrayList<>();
+        for (Person person : people) {
+            if (!deletePerson(person)) {
+                restorePeople(deletePeople);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * This function is used to restore list people if delete process false
+     *
+     * @param deletePeople
+     */
+    private void restorePeople(List<Person> deletePeople) {
+        people.addAll(deletePeople);
+    }
+
+    /**
+     * This function is used to delete one person
+     *
+     * @param person
```

Comment:
please explain where is this param come from, what is value of it

Reply:
Đã update